new role `google_cloud_vpc`

privateip opened the pull request:
This commit adds a new role to the collection for creating google cloud
vpc networks.  This role allows the vpc to be managed as well as any
firewall rules.

Commit:
new role `google_cloud_vpc`

This commit adds a new role to the collection for creating google cloud
vpc networks.  This role allows the vpc to be managed as well as any
firewall rules.

Commit:
fix lint exposed errors

mtraynham commented on network/roles/google_cloud_vpc/defaults/main.yaml:
```diff
@@ -0,0 +1,16 @@
+gcp_network_name: "vpc-{{ id[0:6] }}"
+gcp_network_routing_mode: REGIONAL
+gcp_network_auto_create_subnets: True
+gcp_network_cidr: null
```
Didn't see where `gcp_network_cidr` is being used, wondering if it can be removed?

mtraynham commented on network/roles/google_cloud_vpc/tasks/absent.yaml:
```diff
@@ -0,0 +1,45 @@
+    name: "{{ item.name }}"
+    allowed: "{{ item.allowed }}"
+    source_ranges: "{{ item.source_ranges | default(omit) }}"
+    network: { selfLink: "{{ gcp_network.selfLink }}" }
```
Curious, how is `gcp_network` (and some other variables) defined here?  I'm guessing its external, just trying to understand the flow.

mtraynham reviewed (approved):
Overall LGTM, but I had two questions.

Commit:
removes configurable defaults

This commit changes the role implementation to autogenerate the values
to create google cloud networks.  This role can now run idempontently
with no values being configured.

Commit:
fix for gcp_subnets

This fixes a bug with iterating over gcp subnets and removing them
